fix(core): log error instead of warning for unknown properties and elements

crisbeto opened the pull request:
Changes the Ivy unknown element/property messages from being logged with `console.warn` to `console.error`. This should make them a bit more visible without breaking existing apps. Furthermore, a lot of folks filter out warning messages in the dev tools' console, whereas errors are usually still shown.

Fixes #35699.

AndrewKushnir reviewed (approved):
LGTM, thanks @crisbeto 👍

AndrewKushnir commented:
[Presubmit](http://test/OCL:298390658:BASE:298390707:1583174747011:2b250e59)

pkozlowski-opensource reviewed (approved):
LGTM, just one nit - could you please add a reference to the existing GH issue in the commit message (Fixes #35699)? Thnx!

crisbeto committed:
fix(core): log error instead of warning for unknown properties and elements

Changes the Ivy unknown element/property messages from being logged with `console.warn` to `console.error`. This should make them a bit more visible without breaking existing apps. Furthermore, a lot of folks filter out warning messages in the dev tools' console, whereas errors are usually still shown.

Fixes #35699.

crisbeto commented:
Done.

kara reviewed (approved):
LGTM

AndrewKushnir commented:
MERGE ASSISTANCE: there is a couple test targets failing in g3 with the changes from this PR. The test relies on the fact that `console.warn` is used and we change it to `console.error` instead. The tests are owner by Angular team and I'll create a followup CL that we should land once this change is submitted in g3. Thank you.

IgorMinar commented:
as discussed let's roll this back because it could be breaking e2e tests that assert that there are no errors in the console after the assertions have run. we can readd this in v10.

FDIM commented:
@IgorMinar How come this behavior was changed to warn in angular 9/ivy? It was an error in 8
I just noticed that unit tests are resulting in false-positives when not everything is mocked.
